pkg/platform/kube/apigatewayres/lazy.go

```diff
@@ -78,6 +81,13 @@ func (lc *lazyClient) CreateOrUpdate(ctx context.Context, apiGateway *nuclioio.N
 		return nil, errors.Wrap(err, "Api gateway spec validation failed")
 	}
 
+	// restore scrubbed data
+	if restoredAPIGatewayConfig, err := lc.scrubber.RestoreAPIGatewayConfig(ctx, &platform.APIGatewayConfig{Spec: apiGateway.Spec}); err != nil {
+		return nil, errors.Wrap(err, "Failed to restore scrubber api gateway config")
+	} else {
+		apiGateway.Spec = restoredAPIGatewayConfig.Spec
+	}
```

Comment:
Question - is the secret really needed restoring the spec here?
The controller will create the ingress and then the secret won't be used anymore, correct?

Reply:
@TomerShor how else then we can create a nginx ingress? 
 Imagine we created API Gateway and scrubber it's password. Controller catches this changes and tries to create a nginx ingress. Without restoring we will create ingress with password `$ref:/...` instead if real password then

Reply:
Of course, Sorry I missed an important word in my question lol:

is the secret really needed **after** restoring the spec here?

Reply:
@TomerShor that's a good question. 

yes, we need it, because within update flow we enrich existed api gateway config.

Reply:
So on update do you restore the config, update the data then update the secret?
Or just take the new data in the update request regardless?

Reply:
@TomerShor get existing gw -> restore from secret -> enrich and validate with the new one -> scrub the new one